Source/Bogus.Tests/DataSetTests/DateTest.cs

```diff
@@ -352,5 +358,267 @@ public void can_get_timezone_string()
+
+            sample.Should().Be(windowStart);
+         }
+      }
+
+      [FactWhenDaylightSavingsSupported]
+      public void works_when_range_end_is_exactly_daylight_savings_transition_window_end()
+      {
+         // Arrange
+         var faker = new Faker();
+
+         faker.Random = new Randomizer(localSeed: 5);
+
+         var dstRules = TimeZoneInfo.Local.GetAdjustmentRules();
+
+         var now = DateTime.Now;
+
+         var effectiveRule = dstRules.Single(rule => (rule.DateStart <= now) && (rule.DateEnd >= now));
+
+         var transitionStartTime = CalculateTransitionDateTime(now, effectiveRule.DaylightTransitionStart);
+         var transitionEndTime = transitionStartTime + effectiveRule.DaylightDelta;
+
+         var windowStart = transitionStartTime.AddMinutes(5);
+         var windowEnd = transitionEndTime;
+
+         // Act & Assert
+         for (int i = 0; i < 10000; i++)
+         {
+            var sample = faker.Date.Between(windowStart, windowEnd);
+
+            sample.Should().Be(windowEnd);
+         }
+      }
+
+      [FactWhenDaylightSavingsSupported]
+      public void works_when_start_time_is_invalid_due_to_DST_change_window()
+      {
+         // Arrange
+         var faker = new Faker();
+
+         faker.Random = new Randomizer(localSeed: 5);
+
+         var dstRules = TimeZoneInfo.Local.GetAdjustmentRules();
+
+         var now = DateTime.Now;
+
+         var effectiveRule = dstRules.Single(rule => (rule.DateStart <= now) && (rule.DateEnd >= now));
+
+         var transitionStartTime = CalculateTransitionDateTime(now, effectiveRule.DaylightTransitionStart);
+         var transitionEndTime = transitionStartTime + effectiveRule.DaylightDelta;
+
+         var date1 = new DateTimeOffset(transitionEndTime.AddMinutes(-1), TimeZoneInfo.Local.BaseUtcOffset);
+         var date2 = new DateTimeOffset(transitionEndTime, TimeZoneInfo.Local.BaseUtcOffset + effectiveRule.DaylightDelta);
+
+         // Act
+         var sample = faker.Date.BetweenOffset(date1, date2);
+
+         // Assert
+         _testOutput.WriteLine("BetweenOffset result: {0}", sample);
+
+         sample.Should().Be(transitionEndTime);
+      }
+
+      private DateTime CalculateTransitionDateTime(DateTime now, TimeZoneInfo.TransitionTime transition)
+      {
+         // Based on code found at: https://docs.microsoft.com/en-us/dotnet/api/system.timezoneinfo.transitiontime.isfixeddaterule
+
+         if (transition.IsFixedDateRule)
+         {
+            return new DateTime(
+               now.Year,
+               transition.Month,
+               transition.Day,
+               transition.TimeOfDay.Hour,
+               transition.TimeOfDay.Minute,
+               transition.TimeOfDay.Second,
+               transition.TimeOfDay.Millisecond,
+               DateTimeKind.Local);
+         }
+
+         var calendar = CultureInfo.CurrentCulture.Calendar;
+
+         var startOfWeek = transition.Week * 7 - 6;
```

Comment:
Again, I still have the question here if `7` is always correct for week-lengths.

Reply:
As far as I can tell, it is always correct _in .NET specifically_. Consider the following documentation, for instance:

> CalendarWeekRule Enum (System.Globalization)
> The first week based on the FirstDay value can have one to seven days. The first week based on the FirstFullWeek value **always has seven days**. The first week ...

([link](https://learn.microsoft.com/en-us/dotnet/api/system.globalization.calendarweekrule?view=net-6.0))

...and the fact that the `DayOfWeek` enum has _exactly_ the western 7 days of the week in it.

Reply:
Good to know, thanks!